bugbug/repository.py

```diff
@@ -1552,6 +1553,37 @@ def trigger_pull() -> None:
     trigger_pull()
 
 
+def generate_commit_from_raw_patch(
+    repo_dir: str, base_rev: str, patch: bytes, commit_msg: str
+):
+    # Open the hg repository with minimal extensions to avoid compatibility issues
+    configs = [
+        b"extensions.evolve=!",  # Disable evolve extension
+        b"extensions.mozext=!",  # Disable mozext extension
+    ]
+    with hglib.open(repo_dir, configs=configs) as hg:
+        logger.info("Updating to base revision %s...", base_rev)
+        hg.update(base_rev.encode("utf-8"), clean=True)
+
+        logger.info("Applying the patch ...")
+        hg.import_(
+            patches=io.BytesIO(patch),
+            message=commit_msg.encode("utf-8"),
+            user="bugbot <bugbug@mozilla.tld>".encode("utf-8"),
```

Comment:
We might need the actual user, I think there are some features based on author experience